spec/controllers/trainer_controller_spec.rb

```diff
@@ -0,0 +1,228 @@
+        card = check_review_card(card, 'car', 3)
+        expect(card.review_step).to eq(1)
+      end
+
+      it 'set review_count try=4' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 3)
+        expect(card.review_attempt).to eq(2)
+      end
+    end
+
+    describe 'incorrect and correct translation' do
+      before do
+        @user = create(:user)
+        @block = create(:block, user: @user)
+        @controller.send(:auto_login, @user)
+      end
+
+      it 'set review_date try=3' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 1)
+        card = check_review_card(card, 'house', 1)
+        expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
+            to eq((Time.zone.now + 7.days).strftime('%Y-%m-%d %H:%M'))
+      end
+
+      it 'set review_step try=3' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 1)
+        card = check_review_card(card, 'house', 1)
+        expect(card.review_step).to eq(4)
+      end
+
+      it 'set review_count try=3' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 1)
+        card = check_review_card(card, 'house', 1)
+        expect(card.review_attempt).to eq(1)
+      end
+
+      it 'set review_date try=4' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 2)
+        card = check_review_card(card, 'house', 1)
+        expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
+            to eq((Time.zone.now + 12.hours).strftime('%Y-%m-%d %H:%M'))
+      end
+
+      it 'set review_step try=4' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 2)
+        card = check_review_card(card, 'house', 1)
+        expect(card.review_step).to eq(2)
+      end
+
+      it 'set review_count try=4' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 2)
+        card = check_review_card(card, 'house', 1)
+        expect(card.review_attempt).to eq(1)
+      end
+
+      it 'set review_date try=4+1' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 2)
+        card = check_review_card(card, 'house', 1)
+        card = create_and_check_review_card(@user, @block, 1, 'house')
+        expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
+            to eq((Time.zone.now + 12.hours).strftime('%Y-%m-%d %H:%M'))
+      end
+
+      it 'set review_step try=4+1' do
+        card = create_and_check_review_card(@user, @block, 3, 'car')
+        card = check_review_card(card, 'car', 2)
+        card = check_review_card(card, 'house', 1)
```

Comment:
Useless assignment to variable - `card`.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.